openhands/controller/agent_controller.py

```diff
@@ -383,7 +383,10 @@ async def _step(self) -> None:
 
         if self.delegate is not None:
             assert self.delegate != self
-            await self._delegate_step()
+            if self.delegate.get_agent_state() == AgentState.PAUSED:
+                await asyncio.sleep(1)
+            else:
+                await self._delegate_step()
```

Comment:
Right, it is spammy. Alternatively, we can kill this logger.debug() from the delegate. Didn't we clean it up before? ... 😇

Reply:
Inside `_delegate_step` the flow is a bit different and the logs may make sense there (for now). I think with this small change a lot is prevented (if paused).